app/Spago.hs

```diff
@@ -250,21 +281,55 @@ listPackages = do
 sources :: IO ()
 sources = do
   config <- Config.ensureConfig
-  let
-    deps = getAllDependencies config
-    globs = getGlobs deps
-  _ <- traverse echo globs
+  deps <- getProjectDeps config
+  _ <- traverse echo $ getGlobs deps
   pure ()
 
 
+verify :: Maybe Int -> Maybe PackageName -> IO ()
+verify maybeLimit maybePackage = do
+  Config{..} <- Config.ensureConfig
+  case maybePackage of
+    -- If no package is specified, verify all of them
+    Nothing -> verifyPackages packages (Map.toList packages)
+    -- In case we have a package, search in the package set for it
+    Just packageName -> do
+      case Map.lookup packageName packages of
+        Nothing -> die $ "No packages found with the name " <> Text.pack (show packageName)
+        Just package -> do
+          reverseDeps <- getReverseDeps packages packageName
+          let toVerify = [(packageName, package)] <> reverseDeps
+          verifyPackages packages toVerify
+  where
+    verifyPackages :: PackageSet -> [(PackageName, Package)] -> IO ()
+    verifyPackages packageSet packages = do
+      echoStr $ "Verifying "
+              <> show (length packages)
+              <> " packages, this might take a while.."
+      traverse_ (verifyPackage packageSet) (fst <$> packages)
+
+    verifyPackage :: PackageSet -> PackageName -> IO ()
```

Comment:
i don't quite understand why this should be nested in here instead of being top level? it's actually bothered me that we have `verify` to verify an entire set instead of a given specific package

Reply:
Are the use-cases for `verify $package` and `verify-set` similar?

Reply:
almost not at all, because you will verify-set locally but only run verify in CI

Reply:
I.e. `verify-set` is to verify the consistency of the whole package-set right?

Reply:
Funny thing is that it might not, since some packages might be orphans with no other shared references. But yes, overall it's meant to do that, whereas verifying a package does the reverse deps/referrers check because you want to make sure the it doesn't break them without having to check the whole set.

Reply:
Thanks for the explanation 👍 Let's go with `verify-set` then 🙌